// bosta/lib/bosta/deliveries/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var Delivery = /*#__PURE__*/function () {
  function Delivery(apiClient) {
    (0, _classCallCheck2["default"])(this, Delivery);
    this.apiClient = apiClient;
  }

  (0, _createClass2["default"])(Delivery, [{
    key: "createDelivery",
    value: function () {
      var _createDelivery = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(type, specs, cod, dropOffAddress, businessReference, receiver, notes) {
        var data, result;
        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                _context.prev = 0;
                data = {
                  type: type,
                  specs: specs,
                  cod: cod,
                  dropOffAddress: dropOffAddress,
                  businessReference: businessReference,
                  receiver: receiver,
                  notes: notes ? notes : undefined
                };
                _context.next = 4;
                return this.apiClient.send('post', 'deliveries', data);

              case 4:
                result = _context.sent;

                if (!(result.success = true)) {
                  _context.next = 9;
                  break;
                }

                return _context.abrupt("return", result.data || result.message);

              case 9:
                throw new Error(result.message);

              case 10:
                _context.next = 15;
                break;

              case 12:
                _context.prev = 12;
                _context.t0 = _context["catch"](0);
                throw new Error(_context.t0);

              case 15:
              case "end":
                return _context.stop();
            }
          }
        }, _callee, this, [[0, 12]]);
      }));

      function createDelivery(_x, _x2, _x3, _x4, _x5, _x6, _x7) {
        return _createDelivery.apply(this, arguments);
      }

      return createDelivery;
    }()
  }, {
    key: "getAllDeliveries",
    value: function () {
      var _getAllDeliveries = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
        var result;
        return _regenerator["default"].wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                _context2.prev = 0;
                _context2.next = 3;
                return this.apiClient.send('get', 'deliveries');

              case 3:
                result = _context2.sent;

                if (!(result.success = true)) {
                  _context2.next = 8;
                  break;
                }

                return _context2.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context2.next = 14;
                break;

              case 11:
                _context2.prev = 11;
                _context2.t0 = _context2["catch"](0);
                throw new Error(_context2.t0);

              case 14:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2, this, [[0, 11]]);
      }));

      function getAllDeliveries() {
        return _getAllDeliveries.apply(this, arguments);
      }

      return getAllDeliveries;
    }()
  }, {
    key: "getDeliveryByTrakingNumber",
    value: function () {
      var _getDeliveryByTrakingNumber = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(trackingNumber) {
        var result;
        return _regenerator["default"].wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                _context3.prev = 0;
                _context3.next = 3;
                return this.apiClient.send('get', "deliveries/".concat(trackingNumber));

              case 3:
                result = _context3.sent;

                if (!(result.success = true)) {
                  _context3.next = 8;
                  break;
                }

                return _context3.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context3.next = 14;
                break;

              case 11:
                _context3.prev = 11;
                _context3.t0 = _context3["catch"](0);
                throw new Error(_context3.t0);

              case 14:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3, this, [[0, 11]]);
      }));

      function getDeliveryByTrakingNumber(_x8) {
        return _getDeliveryByTrakingNumber.apply(this, arguments);
      }

      return getDeliveryByTrakingNumber;
    }()
  }, {
    key: "trackDelivery",
    value: function () {
      var _trackDelivery = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(trackingNumber) {
        var result;
        return _regenerator["default"].wrap(function _callee4$(_context4) {
          while (1) {
            switch (_context4.prev = _context4.next) {
              case 0:
                _context4.prev = 0;
                _context4.next = 3;
                return this.apiClient.send('get', "deliveries/".concat(trackingNumber, "/tracking"));

              case 3:
                result = _context4.sent;

                if (!(result.success = true)) {
                  _context4.next = 8;
                  break;
                }

                return _context4.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context4.next = 14;
                break;

              case 11:
                _context4.prev = 11;
                _context4.t0 = _context4["catch"](0);
                throw new Error(_context4.t0);

              case 14:
              case "end":
                return _context4.stop();
            }
          }
        }, _callee4, this, [[0, 11]]);
      }));

      function trackDelivery(_x9) {
        return _trackDelivery.apply(this, arguments);
      }

      return trackDelivery;
    }()
  }, {
    key: "printDeliveryAWB",
    value: function () {
      var _printDeliveryAWB = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(deliveryId) {
        var result;
        return _regenerator["default"].wrap(function _callee5$(_context5) {
          while (1) {
            switch (_context5.prev = _context5.next) {
              case 0:
                _context5.prev = 0;
                _context5.next = 3;
                return this.apiClient.send('get', "deliveries/awb/".concat(deliveryId));

              case 3:
                result = _context5.sent;

                if (!(result.success = true)) {
                  _context5.next = 8;
                  break;
                }

                return _context5.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context5.next = 14;
                break;

              case 11:
                _context5.prev = 11;
                _context5.t0 = _context5["catch"](0);
                throw new Error(_context5.t0);

              case 14:
              case "end":
                return _context5.stop();
            }
          }
        }, _callee5, this, [[0, 11]]);
      }));

      function printDeliveryAWB(_x10) {
        return _printDeliveryAWB.apply(this, arguments);
      }

      return printDeliveryAWB;
    }()
  }, {
    key: "updateDelivery",
    value: function () {
      var _updateDelivery = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6(deliveryId, updatePayload) {
        var result;
        return _regenerator["default"].wrap(function _callee6$(_context6) {
          while (1) {
            switch (_context6.prev = _context6.next) {
              case 0:
                _context6.prev = 0;
                _context6.next = 3;
                return this.apiClient.send('put', "deliveries/".concat(deliveryId), updatePayload);

              case 3:
                result = _context6.sent;

                if (!(result.success = true)) {
                  _context6.next = 8;
                  break;
                }

                return _context6.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context6.next = 14;
                break;

              case 11:
                _context6.prev = 11;
                _context6.t0 = _context6["catch"](0);
                throw new Error(_context6.t0);

              case 14:
              case "end":
                return _context6.stop();
            }
          }
        }, _callee6, this, [[0, 11]]);
      }));

      function updateDelivery(_x11, _x12) {
        return _updateDelivery.apply(this, arguments);
      }

      return updateDelivery;
    }()
  }, {
    key: "terminateDelivery",
    value: function () {
      var _terminateDelivery = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee7(deliveryId) {
        var result;
        return _regenerator["default"].wrap(function _callee7$(_context7) {
          while (1) {
            switch (_context7.prev = _context7.next) {
              case 0:
                _context7.prev = 0;
                _context7.next = 3;
                return this.apiClient.send('delete', "deliveries/".concat(deliveryId));

              case 3:
                result = _context7.sent;

                if (!(result.success = true)) {
                  _context7.next = 8;
                  break;
                }

                return _context7.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context7.next = 14;
                break;

              case 11:
                _context7.prev = 11;
                _context7.t0 = _context7["catch"](0);
                throw new Error(_context7.t0);

              case 14:
              case "end":
                return _context7.stop();
            }
          }
        }, _callee7, this, [[0, 11]]);
      }));

      function terminateDelivery(_x13) {
        return _terminateDelivery.apply(this, arguments);
      }

      return terminateDelivery;
    }()
  }]);
  return Delivery;
}();

var _default = Delivery;
exports["default"] = _default;
