// bosta/lib/bosta/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _axios = _interopRequireDefault(require("axios"));

var _index = _interopRequireDefault(require("./pickups/index.js"));

var _index2 = _interopRequireDefault(require("./deliveries/index.js"));

var _cities = _interopRequireDefault(require("./cities"));

var _zones = _interopRequireDefault(require("./zones"));

var _utils = require("../utils");

var Bosta = /*#__PURE__*/function () {
  function Bosta(apiKey, baseUrl) {
    (0, _classCallCheck2["default"])(this, Bosta);
    this.apiKey = apiKey;
    this.baseUrl = baseUrl ? baseUrl : 'https://app.bosta.co';
    this.deliveryStates = _utils.deliveryStates;
    this.deliveryTypes = _utils.deliveryTypes;
    this.pickupStates = _utils.pickupStates;
    this.pickupTimeSlots = _utils.pickupTimeSlots;
    this.pickup = new _index["default"](this);
    this.delivery = new _index2["default"](this);
    this.city = new _cities["default"](this);
    this.zone = new _zones["default"](this);
  }

  (0, _createClass2["default"])(Bosta, [{
    key: "send",
    value: function () {
      var _send = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(method, path, body) {
        var result;
        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                _context.prev = 0;
                _context.next = 3;
                return (0, _axios["default"])({
                  method: method,
                  url: "".concat(this.baseUrl, "/api/v0/").concat(path),
                  data: body ? body : undefined,
                  headers: {
                    'Authorization': this.apiKey,
                    'contentType': 'application/json',
                    'X-Requested-By': 'nodejs-sdk'
                  },
                  timeout: 30000
                });

              case 3:
                result = _context.sent;
                return _context.abrupt("return", result);

              case 7:
                _context.prev = 7;
                _context.t0 = _context["catch"](0);
                throw new Error(_context.t0.response.data.message);

              case 10:
              case "end":
                return _context.stop();
            }
          }
        }, _callee, this, [[0, 7]]);
      }));

      function send(_x, _x2, _x3) {
        return _send.apply(this, arguments);
      }

      return send;
    }()
  }]);
  return Bosta;
}();

var _default = Bosta;
exports["default"] = _default;
