// bosta/lib/bosta/pickups/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var Pickup = /*#__PURE__*/function () {
  function Pickup(apiClient) {
    (0, _classCallCheck2["default"])(this, Pickup);
    this.apiClient = apiClient;
  }

  (0, _createClass2["default"])(Pickup, [{
    key: "getBusinessPickupLocations",
    value: function () {
      var _getBusinessPickupLocations = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
        var result;
        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                _context.prev = 0;
                _context.next = 3;
                return this.apiClient.send('get', 'pickup-locations');

              case 3:
                result = _context.sent;

                if (!(result.success = true)) {
                  _context.next = 8;
                  break;
                }

                return _context.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context.next = 14;
                break;

              case 11:
                _context.prev = 11;
                _context.t0 = _context["catch"](0);
                throw new Error(_context.t0);

              case 14:
              case "end":
                return _context.stop();
            }
          }
        }, _callee, this, [[0, 11]]);
      }));

      function getBusinessPickupLocations() {
        return _getBusinessPickupLocations.apply(this, arguments);
      }

      return getBusinessPickupLocations;
    }()
  }, {
    key: "createPickup",
    value: function () {
      var _createPickup = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(businessLocationId, scheduledDate, scheduledTimeSlot, contactPerson, notes) {
        var data, result;
        return _regenerator["default"].wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                _context2.prev = 0;
                data = {
                  businessLocationId: businessLocationId,
                  scheduledDate: scheduledDate,
                  scheduledTimeSlot: scheduledTimeSlot,
                  contactPerson: contactPerson,
                  notes: notes ? notes : undefined
                };
                _context2.next = 4;
                return this.apiClient.send('post', 'pickups', data);

              case 4:
                result = _context2.sent;

                if (!(result.success = true)) {
                  _context2.next = 9;
                  break;
                }

                return _context2.abrupt("return", result.data || result.message);

              case 9:
                throw new Error(result.message);

              case 10:
                _context2.next = 15;
                break;

              case 12:
                _context2.prev = 12;
                _context2.t0 = _context2["catch"](0);
                throw new Error(_context2.t0);

              case 15:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2, this, [[0, 12]]);
      }));

      function createPickup(_x, _x2, _x3, _x4, _x5) {
        return _createPickup.apply(this, arguments);
      }

      return createPickup;
    }()
  }, {
    key: "getAllPickups",
    value: function () {
      var _getAllPickups = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3() {
        var result;
        return _regenerator["default"].wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                _context3.prev = 0;
                _context3.next = 3;
                return this.apiClient.send('get', 'pickups');

              case 3:
                result = _context3.sent;

                if (!(result.success = true)) {
                  _context3.next = 8;
                  break;
                }

                return _context3.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context3.next = 14;
                break;

              case 11:
                _context3.prev = 11;
                _context3.t0 = _context3["catch"](0);
                throw new Error(_context3.t0);

              case 14:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3, this, [[0, 11]]);
      }));

      function getAllPickups() {
        return _getAllPickups.apply(this, arguments);
      }

      return getAllPickups;
    }()
  }, {
    key: "getPickupById",
    value: function () {
      var _getPickupById = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(pickupId) {
        var result;
        return _regenerator["default"].wrap(function _callee4$(_context4) {
          while (1) {
            switch (_context4.prev = _context4.next) {
              case 0:
                _context4.prev = 0;
                _context4.next = 3;
                return this.apiClient.send('get', "pickups/".concat(pickupId));

              case 3:
                result = _context4.sent;

                if (!(result.success = true)) {
                  _context4.next = 8;
                  break;
                }

                return _context4.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context4.next = 14;
                break;

              case 11:
                _context4.prev = 11;
                _context4.t0 = _context4["catch"](0);
                throw new Error(_context4.t0);

              case 14:
              case "end":
                return _context4.stop();
            }
          }
        }, _callee4, this, [[0, 11]]);
      }));

      function getPickupById(_x6) {
        return _getPickupById.apply(this, arguments);
      }

      return getPickupById;
    }()
  }, {
    key: "updatePickup",
    value: function () {
      var _updatePickup = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(pickupId, updatePayload) {
        var result;
        return _regenerator["default"].wrap(function _callee5$(_context5) {
          while (1) {
            switch (_context5.prev = _context5.next) {
              case 0:
                _context5.prev = 0;
                _context5.next = 3;
                return this.apiClient.send('put', "pickups/".concat(pickupId), updatePayload);

              case 3:
                result = _context5.sent;

                if (!(result.success = true)) {
                  _context5.next = 8;
                  break;
                }

                return _context5.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context5.next = 14;
                break;

              case 11:
                _context5.prev = 11;
                _context5.t0 = _context5["catch"](0);
                throw new Error(_context5.t0);

              case 14:
              case "end":
                return _context5.stop();
            }
          }
        }, _callee5, this, [[0, 11]]);
      }));

      function updatePickup(_x7, _x8) {
        return _updatePickup.apply(this, arguments);
      }

      return updatePickup;
    }()
  }, {
    key: "deletePickup",
    value: function () {
      var _deletePickup = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6(pickupId) {
        var result;
        return _regenerator["default"].wrap(function _callee6$(_context6) {
          while (1) {
            switch (_context6.prev = _context6.next) {
              case 0:
                _context6.prev = 0;
                _context6.next = 3;
                return this.apiClient.send('delete', "pickups/".concat(pickupId));

              case 3:
                result = _context6.sent;

                if (!(result.success = true)) {
                  _context6.next = 8;
                  break;
                }

                return _context6.abrupt("return", result.data || result.message);

              case 8:
                throw new Error(result.message);

              case 9:
                _context6.next = 14;
                break;

              case 11:
                _context6.prev = 11;
                _context6.t0 = _context6["catch"](0);
                throw new Error(_context6.t0);

              case 14:
              case "end":
                return _context6.stop();
            }
          }
        }, _callee6, this, [[0, 11]]);
      }));

      function deletePickup(_x9) {
        return _deletePickup.apply(this, arguments);
      }

      return deletePickup;
    }()
  }]);
  return Pickup;
}();

var _default = Pickup;
exports["default"] = _default;
